lib/charms/parca_k8s/v0/parca_scrape.py

```diff
@@ -382,53 +359,13 @@ def __init__(self, charm: ops.CharmBase, relation_name: str = DEFAULT_RELATION_N
             RelationInterfaceMismatchError: The relation with the same name as provided
                 via `relation_name` argument does not have the `parca_scrape` relation
                 interface.
-            ops.RelationRoleMismatchError: If the relation with the same name as provided
+            RelationRoleMismatchError: If the relation with the same name as provided
                 via `relation_name` argument does not have the `ops.RelationRole.requires`
                 role.
         """
-        _validate_relation_by_interface_and_direction(
-            charm, relation_name, RELATION_INTERFACE_NAME, ops.RelationRole.requires
-        )
-
-        super().__init__(charm, relation_name)
-        self._charm = charm
         self._relation_name = relation_name
-        events = self._charm.on[relation_name]
-        self.framework.observe(
-            events.relation_changed, self.on_profiling_provider_relation_changed
-        )
-        self.framework.observe(
-            events.relation_departed, self._on_profiling_provider_relation_departed
-        )
```

Comment:
I'm torn on removing these.  

On one hand I like the idea that libs don't just "operate" your charm and instead they only do things when you ask.  On the other hand, especially for complex relations, there's something nice about abstracting away the details of the low-level relation events and instead providing a higher level "the library says the charm should care now" event like here with `TargetsChangedEvent`.  

Admittedly that is less important here (where we essentially just echo a few relation events), but sometimes it is really helpful.  Are there cases where this is a problem?  Conceptually, I see a compelling benefit in maintaining a clear separation of concerns (lib handles relation events, charm handles lib events) and I don't see a real cost.

Reply:
I'm thinking of something like KubernetesServicePatch.  I think most people using it would say the fact that it "just works" was its best attribute.  No need for a charmer to decide which events should trigger it